src/components/KCopy/KCopy.vue

```diff
@@ -0,0 +1,263 @@
+    */
+  badge: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+    * False if `badge` is true
+    */
+  monospace: {
+    type: Boolean,
+    default: true,
+  },
+  /**
+    * Whether or not the text should be truncated
+    */
+  truncate: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+    * Tooltip text displayed on successful copy
+    */
+  successTooltip: {
+    type: String,
+    default: 'Copied',
+  },
+  /**
+    * Number of characters to truncate at
+    */
+  truncationLimit: {
+    type: Number,
+    default: 8,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'copy', val: string): void
+}>()
+
+const hasSuccessTooltip = computed((): boolean => !!(props.copyTooltip && props.successTooltip))
+const tooltipText = ref(props.copyTooltip)
+
+const badgeTextLabel = computed((): string => {
+  if (!props.badgeText) {
+    return ''
+  }
+
+  // strip trailing colon from label if one exists
+  return props.badgeText.replace(/:$/, '')
+})
+
+const iconTitle = computed(() => {
+  if (props.badgeText) {
+    return `Copy ${badgeTextLabel.value}`
+  }
+  return 'Copy'
+})
+
+const truncateLimitText = computed((): string | null => props.truncate ? `${props.text.substring(0, props.truncationLimit) + '...'}` : null)
+
+// Computed for all dynamic classes
+const truncateStyles = computed((): string | null => props.truncate ? 'copy-element' : null)
+const badgeStyles = computed((): string | null => props.badge ? 'badge-styles' : null)
+const truncateContent = computed((): string | null => props.truncate ? 'truncate-content' : null)
+const useMono = computed((): string | null => props.badge ? null : 'monospace')
+
+const textFormat = computed(() => {
+  if (props.format === 'redacted') {
+    return '*****'
+  } else if (props.format === 'deleted') {
+    return `*${props.text.substring(0, 5)}`
+  }
+  return (props.truncate && props.truncationLimit && truncateLimitText.value) ? truncateLimitText.value.replace(/^"(.*)"$/, '$1') : props.text
+})
+
+const updateTooltipText = (msg: string): void => {
+  tooltipText.value = msg
+  setTimeout(() => {
+    tooltipText.value = props.copyTooltip
+  }, 1800)
+}
+
+const copyIdToClipboard = (executeCopy: (prop: string) => boolean) => {
+  if (!executeCopy(props.text)) {
+    if (hasSuccessTooltip.value) {
+      updateTooltipText('Failed to copy')
+    }
+
+    return
+  }
+
+  if (hasSuccessTooltip.value) {
+    updateTooltipText(props.successTooltip)
+    emit('copy', props.text)
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+.k-copy {
+  align-items: center;
+  display: flex;
+
+  &-badge-text {
+    margin-right: $kui-space-40;
+  }
+
+  .copy-element {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+
+    .truncate-content {
+      display: inline-block;
+      overflow: hidden;
+      text-overflow: ellipsis;
+      white-space: nowrap;
+    }
+  }
+
+  .badge-styles {
+    background-color: #f0f4fa;
+    border-color: #d7d8fe;
```

Comment:
```suggestion
    border-color: var(--kui-color-border-decorative-purple, $kui-color-border-decorative-purple);
```

See the comment above